### DISCUSSION TITLE
svelte: should $todo === todo?

### DISCUSSION THREAD

**Discussion by frederikhors** [urql-graphql/urql#1033] [Q&A]
urql version & exchanges:

```json
"@urql/exchange-graphcache": "3.1.3",
"@urql/svelte": "1.0.0",
"graphql": "15.3.0",
"graphql-tag": "2.11.0",
"svelte": "3.29.0",
```

**Steps to reproduce**

1. Start the server opening this page: https://codesandbox.io/s/apollo-server-urql-svelte-crud-h1pcl
1. Open Incognito browser session
1. Go to reproduction: https://b1qpk.csb.app/
1. Click on "Todos list"
1. The list shows
1. Click on the first todo link "Open"
1. The todo shows, `$todo` and `todo` are different
1. Click on "Todos list"
1. Click on the first todo link "Open"
1. The todo shows, `$todo` and `todo` are the same

**Expected behavior**

`$todo` !== `todo`

Am I wrong?

**Actual behavior**

`$todo` and `todo` are the same after after the first time in cache

**Additional**

[CodeSandBox client project](https://codesandbox.io/s/urql-svelte-100-store-and-object-are-the-same-b1qpk?file=/Todo.svelte)
[CodeSandBox server project](https://codesandbox.io/s/apollo-server-urql-svelte-crud-h1pcl)

**Comment by kitten** [comment]
So first of all: `todo` is a special store here, that means that its properties are getters and you'll see the synchronous state of the store at that moments, which can be useful for callbacks and stuff or timers.

But `$todo` is Svelte automatically subscribing to this store, which means it'll change over time. When you call `query` the synchronous state of `todo` updates, but it may sometimes be that `$todo` lags behind by one call, which is Svelte's update subscriber call.
Also the value of `todo` heavily depends on when you log it.

tl;dr do use the reactive `$todo` with Svelte until you get to a situation where you can't; then it'll make sense to use the getter feature :)
